---
title: "Analysis of 1K1K single cell dataset with lucida"
subtitle: '[Yazar, et al. (2022)](https://doi.org/10.1126/science.abf3041)'
author: "Developed by [Gabriel Hoffman](http://gabrielhoffman.github.io/)"
date: "Run on `r format(Sys.time())`"
documentclass: article
vignette: >
  %\VignetteIndexEntry{Analysis of 1K1K single cell dataset with lucida}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  %\usepackage[utf8]{inputenc}
output:
  BiocStyle::html_document:
    toc: true
    toc_float: true
---

<!---

cd /hpc/users/hoffmg01/work/lucida_analysis
module load pandoc

# scp minerva:/hpc/users/hoffmg01/work/lucida_analysis/1k1k_analysis.html .

# rm -rf 1k1k_analysis_cache/

rmarkdown::render("1k1k_analysis.Rmd")


# format H5AD
SRC=/hpc/users/hoffmg01/work/GenomicDataStream_analysis/recode_h5ad.py 
FILE=/sc/arion/projects/CommonMind/hoffman/scRNAseq_data/Yazar_Science_2022/08984b3c-3189-4732-be22-62f1fe8f15a4.h5ad

$SRC --input $FILE --sortBy cell_type,donor_id --format CSC --compression lzf --out $(dirname $FILE)/1k1k_sort_CSC_lzf.h5ad

--->


```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  error = FALSE,
  tidy = FALSE,
  dev = c("png", "pdf"),
  cache = TRUE,
  lazy.cache = FALSE
)
```

```{r packages, cache=FALSE}
library(lucida)
library(SingleCellExperiment)
library(tidyverse)
library(ggplot2)
library(nebula)

thm = theme_classic(15) +
    theme(aspect.ratio = 1, 
      plot.title = element_text(hjust = 0.5)) 
```

# Lucida analysis
```{r load, cache=FALSE}
file = "/sc/arion/projects/CommonMind/hoffman/scRNAseq_data/Yazar_Science_2022/1k1k_sort_CSC_lzf.h5ad"
sce = readH5AD(file)
sce$libSize = sce$nCount_RNA

# filter to include only cell types with 1k observations
tab = table(sce$cell_type)
sce = sce[,sce$cell_type %in% names(tab[tab>1000])]
sce = sce[,sce$libSize > 1000]
sce$cell_type = droplevels(sce$cell_type)
```

```{r lucida}
fit = lucida(sce, ~ age + sex + (1|donor_id), 
  cluster_id = "cell_type")
```

## Plots
```{r plotDispEsts, fig.height = 9, fig.width = 9}
plotDispEsts(fit, ncol=5)
```

```{r plotMA, fig.height = 9, fig.width = 9}
plotMA(fit, coef='age', ncol=5)
plotMA(fit, coef='sexmale', ncol=5)
```

```{r plotVolcano, fig.height = 9, fig.width = 9}
plotVolcano(fit, coef='age', ncol=5)
plotVolcano(fit, coef='sexmale', ncol=5)
```

```{r lucida.null}
fit.null <- lucida(sce, ~ (1|donor_id), 
  cluster_id = "cell_type")
```

```{r vp, fig.height = 9, fig.width = 9}
vp <- fitVarPart(fit, fit.null)

plotVarPart(sortCols(vp), ncol=5)
```

# Results
```{r results}
results(fit, "age") %>%
  group_by(cluster_id) %>%
  summarize( n = n()) %>%
  ggplot(aes(n, cluster_id, fill=cluster_id)) +
    geom_bar(stat="identity") +
    thm +
    theme(aspect.ratio = 2) +
    scale_x_continuous(limits=c(0,NA), expand=c(0,0)) +
    xlab("# genes")

results(fit, "age") %>%
  group_by(cluster_id) %>%
  summarize( nSignif = sum(FDR < 0.05)) %>%
  ggplot(aes(nSignif, cluster_id, fill=cluster_id)) +
    geom_bar(stat="identity") +
    thm +
    theme(aspect.ratio = 2) +
    scale_x_continuous(limits=c(0,NA), expand=c(0,0)) +
    xlab("# significant genes") +
    ggtitle("Age")

results(fit, "sexmale") %>%
  group_by(cluster_id) %>%
  summarize( nSignif = sum(FDR < 0.05)) %>%
  ggplot(aes(nSignif, cluster_id, fill=cluster_id)) +
    geom_bar(stat="identity") +
    thm +
    theme(aspect.ratio = 2) +
    scale_x_continuous(limits=c(0,NA), expand=c(0,0)) +
    xlab("# significant genes") +
    ggtitle("Sex")
```

















