---
title: "Analysis of PsychAD single cell dataset with lucida"
subtitle: '[Lee, et al. (2025)](https://psych-ad.org)'
author: "Developed by [Gabriel Hoffman](http://gabrielhoffman.github.io/)"
date: "Run on `r format(Sys.time())`"
documentclass: article
vignette: >
  %\VignetteIndexEntry{Analysis of PsychAD single cell dataset with lucida}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  %\usepackage[utf8]{inputenc}
output:
  BiocStyle::html_document:
    toc: true
    toc_float: true
---

<!---

cd /hpc/users/hoffmg01/work/lucida_analysis
module load pandoc

# rm -rf PsychAD_analysis_cache/

rmarkdown::render("PsychAD_analysis.Rmd")

# https://docs.google.com/spreadsheets/d/13T9QVoky6bUMDhupfrcXLavOtsLSNWF6/edit?gid=386465612#gid=386465612

# Recode each PsychAD file
SRC=/hpc/users/hoffmg01/work/GenomicDataStream_analysis/recode_h5ad.py 

OUT=/sc/arion/projects/CommonMind/hoffman/scRNAseq_data/psychAD

for FILE in $(ls /sc/arion/projects/psychAD/NPS-AD/freeze2_rc/h5ad_final/*_2024-02*)
do
  OUTFILE=$OUT/$(basename $FILE .h5ad)_sort_CSC_lzf.h5ad
  $SRC --input $FILE --sortBy class,subclass,subtype,Channel,SubID --format CSC --compression lzf --out $OUTFILE
done




--->


```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  error = FALSE,
  tidy = FALSE,
  dev = c("png", "pdf"),
  cache = TRUE,
  lazy.cache = FALSE
)
```

```{r packages, cache=FALSE}
library(lucida)
library(SingleCellExperiment)
library(tidyverse)
library(ggplot2)

thm = theme_classic(15) +
    theme(aspect.ratio = 1, 
      plot.title = element_text(hjust = 0.5)) 
```

# Lucida analysis
```{r load, cache=FALSE}
file = "/sc/arion/projects/CommonMind/hoffman/scRNAseq_data/psychAD/MSSM_2024-02-01_16_17_sort_CSC_lzf.h5ad"
sce = readH5AD(file)
sce$libSize = sce$n_counts
```

```{r lucida}
sce$c02x[sce$c02x == "NA"] <- NA
fit = lucida(sce, ~ c02x + Sex + PMI + (1|SubID), cluster_id = "class")
```

```{r lucida.null}
fit.null <- lucida(sce, ~ (1|SubID), cluster_id = "class")
```

```{r vp, fig.height = 20, fig.width =15}
vp <- fitVarPart(fit, fit.null)

plotVarPart(sortCols(vp), ncol=5)
```




## Plots
```{r plotDispEsts, fig.height = 10, fig.width = 12}
plotDispEsts(fit, ncol=5)
```

```{r plotMA, fig.height = 12, fig.width = 12}
plotMA(fit, coef='age', ncol=5)
plotMA(fit, coef='sexmale', ncol=5)
```

```{r plotVolcano, fig.height = 15, fig.width = 12}
plotVolcano(fit, coef='age', ncol=5)
plotVolcano(fit, coef='sexmale', ncol=5)
```


